disnake/channel.py

```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
+            The new default auto archive duration in minutes for threads created in this channel.
+            Must be one of ``60``, ``1440``, ``4320``, or ``10080``.
+        reason: Optional[:class:`str`]
+            The reason for editing this channel. Shows up on the audit log.
+
+        Raises
+        ------
+        InvalidArgument
+            If position is less than 0 or greater than the number of channels, or if
+            the permission overwrite information is not in proper form.
+        Forbidden
+            You do not have permissions to edit the channel.
+        HTTPException
+            Editing the channel failed.
+
+        Returns
+        -------
+        Optional[:class:`ForumChannel`]
+            The newly edited forum channel. If the edit was only positional
+            then ``None`` is returned instead.
+        """
+        payload = await self._edit(options, reason=reason)
+        if payload is not None:
+            # the payload will always be the proper channel payload
+            return self.__class__(state=self._state, guild=self.guild, data=payload)  # type: ignore
+
+    @utils.copy_doc(disnake.abc.GuildChannel.clone)
+    async def clone(
+        self, *, name: Optional[str] = None, reason: Optional[str] = None
+    ) -> ForumChannel:
+        return await self._clone_impl(
+            {
+                "topic": self.topic,
+                "nsfw": self.nsfw,
+                "default_auto_archive_duration": self.default_auto_archive_duration,
+            },
+            name=name,
+            reason=reason,
+        )
+
+    def get_thread(self, thread_id: int, /) -> Optional[Thread]:
+        """Returns a thread with the given ID.
+
+        Parameters
+        ----------
+        thread_id: :class:`int`
+            The ID to search for.
+
+        Returns
+        -------
+        Optional[:class:`Thread`]
+            The returned thread of ``None`` if not found.
+        """
+        return self.guild.get_thread(thread_id)
+
+    async def fetch_thread(self, thread_id: int, /) -> Thread:
+        """|coro|
+
+        Retrieves a :class:`.Thread` with the given ID.
+
+        .. note::
+
+            This method is an API call. For general usage, consider :meth:`get_thread` instead.
+
+        Raises
+        ------
+        InvalidData
+            An unknown channel type was received from Discord
+            or the guild the channel belongs to is not the same
+            as the one in this object points to.
+        HTTPException
+            Retrieving the channel failed.
+        NotFound
+            Invalid Channel ID.
+        Forbidden
+            You do not have permission to fetch this channel.#
+
+        Returns
+        -------
+        :class:`.Thread`
+            The channel from the ID.
+        """
+        return await self.guild.fetch_channel(thread_id)  # type: ignore
+
+    async def create_thread(
```

Comment:
perhaps this should be named `create_post`?

Reply:
As we discussed, we will keep it as thread https://canary.discord.com/channels/927603005627437097/927603006579568723/962117732264988682